payjoin-io/src/error.rs

```diff
@@ -0,0 +1,61 @@
+#[derive(Debug)]
+pub struct Error(pub(crate) InternalError);
+
+#[derive(Debug)]
+pub(crate) enum InternalError {
+    Ohttp(ohttp::Error),
+    Bhttp(bhttp::Error),
+    ParseUrl(url::ParseError),
+    Ureq(ureq::Error),
+    Io(std::io::Error),
+    JoinError(tokio::task::JoinError),
+}
+
+macro_rules! impl_from_error {
+    ($from:ty, $to:ident) => {
+        impl From<$from> for Error {
+            fn from(value: $from) -> Self { Self(InternalError::$to(value)) }
+        }
+    };
+}
```

Comment:
This is very neat and thought through. However, we should be able to get all of these errors from the payjoin crate instead of re-implementing them.

Reply:
What I mean by this is to encapsulate a payjoin_io::Error veariant PayjoinV2(payjoin::v2::Error)